components/Login.tsx

```diff
@@ -1,54 +1,134 @@
   const mutation = useSendMagicLinkMutation();
 
+  const showUserIcon = useBreakpointValue({ base: false, sm: true });
+
   const handleSubmit = (event: FormEvent) => {
-    event.preventDefault();
-    const formData = new FormData(event.target as HTMLFormElement);
-    const macID = formData.get("macID");
-    if (typeof macID === "string") {
+    if (macID.length > 0) {
       mutation.mutate(macID);
     }
+    event.preventDefault();
   };
 
-  switch (mutation.status) {
-    case "loading":
-      return <p>Loading...</p>;
-    case "success":
-      return <p>Check your email!</p>;
-    case "error":
-      return <p>Something went wrong, please try again later.</p>;
-    case "idle":
-      return (
+  const secondaryTextColor = useColor(Colors.secondaryText);
+
+  return (
+    <Modal isOpen={isOpen} onClose={onClose} size="lg">
+      <ModalOverlay />
+      <ModalContent pt={5} pb={4}>
         <form onSubmit={handleSubmit}>
-          <Flex as="fieldset" gap={2}>
-            <InputGroup>
-              <Input
-                variant="filled"
-                type="text"
-                name="macID"
-                aria-label="macID"
-                required
-                placeholder={macIDPlaceholder}
-              />
-              <InputRightAddon>@mcmaster.ca</InputRightAddon>
-            </InputGroup>
-            <Button variant="solid" type="submit">
-              Login
+          <ModalBody>
+            <HStack align="flex-start" spacing={5}>
+              {showUserIcon && (
+                <Icon
+                  as={UserIcon}
+                  width={12}
+                  padding={3}
+                  height={12}
+                  borderRadius="full"
+                  color="green.600"
+                  backgroundColor="green.100"
+                />
+              )}
+
+              <VStack align="flex-start" spacing={3}>
+                <Heading as="h1" size="md" fontWeight="medium">
+                  Login
+                </Heading>
+                <Text color={secondaryTextColor}>
+                  {(mutation.isIdle || mutation.isLoading) &&
+                    `To create a post you must sign in. Please enter your MacID below.`}
+                  {mutation.isError &&
+                    `Something went wrong, please try again.`}
+                  {mutation.isSuccess &&
+                    `A login link has been sent to ${
+                      macID + EMAIL_SUFFIX
+                    }! Please check your email.`}
+                </Text>
+
+                {(mutation.isIdle || mutation.isLoading) && (
+                  <InputGroup>
+                    <Input
+                      variant="outline"
+                      type="text"
+                      required
+                      autoFocus
+                      autoCapitalize="off"
+                      autoCorrect="off"
+                      autoComplete="username"
+                      spellCheck="false"
```

Comment:
Why do we need to mention the values for autoCapitalize, autoCorrect, spellCheck, and autoComplete? They don't change anything in the UI.

Reply:
They're mostly for mobile and don't change much for desktop

Reply:
I see, that makes sense. Thanks.